### PR TITLE
Faster runtime store build

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Consolidating the individual package references into a separate project and referencing that in the runtime store project reduces the build time ~90%.

### CODE REVIEW

**Comment by natemcmaster** [comment]
I tried to run locally and this fails. Known issue?

> C:\Users\namc\.dotnet\x64\sdk\2.0.0-preview2-006215\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.CrossGen.targets(49,5): error : Unable to find resolved path for 'coreclr'. [C:\dev\aspnet\MetaPackages\src\Microsoft.AspNetCore.RuntimeStore\Microsoft.AspNetCore.RuntimeStore.csproj]

**Comment by JunTaoLuo** [comment]
@natemcmaster yes, we pass in the platform via environment variable. set PACKAGE_CACHE_PLATFORM="x64/x86" and it should work

**Comment by natemcmaster** [comment]
Do you know why it's so much faster? Seems too good to be true, or there's a really bad bug in `dotnet store` targets. I still can't get this to run locally to verify it still works

Also, can we add an error when PACKAGE_CACHE_PLATFORM is undefined?

File: build/repo.targets [hunk 1]
```diff
@@ -64,10 +66,23 @@
       <OutputZipNoTimestamp>$(ArtifactsDir)Build.RS.$(OutputZipSufix).zip</OutputZipNoTimestamp>
     </PropertyGroup>
 
+    <!-- Build reference package -->
+    <Exec Command="dotnet restore" WorkingDirectory="$(RuntimeStoreReferenceDirectory)" />
+    <Exec Command="dotnet pack -c Release" WorkingDirectory="$(RuntimeStoreReferenceDirectory)" />
```
Comment (natemcmaster) [inline build/repo.targets]: FYI - you don't have to shell out to dotnet.exe to run this target.
```xml
<MSBuild Projects="$(RuntimeStoreReferenceDirectory)\Build.RuntimeStore.References.csproj"
    Targets="Pack"
    Properties="Configuration=Release" />
```

**Pull request merged**